sorting-verify/insertion-sort.js

```diff
@@ -1,31 +1,24 @@
-function leftMoving(newArr, n){
-   if(newArr[n]<newArr[n-1]){
-     var temp = newArr[n-1];
-     newArr[n-1] = newArr[n];
-     newArr[n] = temp;
-
-     do {
-       n = n-1;
-       leftMoving(newArr, n);
-     } while (n>1);
+function swap(arr, m, n) {
+  var temp = arr[m];
+  arr[m] = arr[n];
+  arr[n] = temp;
+}
 
-   }
-   return newArr;
+function compareSwapLeft(newArr, n){
+  if(n > 0) {
+    if(newArr[n]<newArr[n-1]){
+      swap(newArr, n, n-1)
+    }
+    compareSwapLeft(newArr, n - 1);
+  }
  }
 
 function insertionSort(arr){
-  var sortedArr = [];
-
-  if (arr.length <= 1){
-    return arr;
-  } else{
-      var newArr = arr;
-      for (var n=0; n<newArr.length; n++){
-        sortedArr = leftMoving(newArr, n);
-      }
-      return sortedArr;
+    var newArr = arr.slice(); //make a new copy
+    for (var n=0; n<newArr.length; n++){
+      compareSwapLeft(newArr, n);
     }
```

Comment:
when define a new variable, always put it in the same scope.

Don't temper the old variable, operate on a new one.